launcher/sdw_updater_gui/Updater.py

```diff
@@ -430,6 +430,65 @@ def _safely_start_vm(vm):
         sdlog.error(str(e))
 
 
+def should_launch_updater(interval):
+    sdlog.info("Starting SecureDrop Launcher")
+
+    status = read_dom0_update_flag_from_disk(with_timestamp=True)
+
+    if _valid_status(status):
+        if _interval_expired(interval, status):
+            sdlog.info("Update interval expired: launching updater.")
+            return True
+        else:
+            if status["status"] == UpdateStatus.UPDATES_OK.value:
+                sdlog.info("Updates OK and interval not expired, launching client.")
+                return False
+            elif status["status"] == UpdateStatus.REBOOT_REQUIRED.value:
+                if last_required_reboot_performed():
+                    sdlog.info(
+                        "Required reboot performed, updating status and launching client."
+                    )
+                    _write_updates_status_flag_to_disk(UpdateStatus.UPDATES_OK)
+                    return False
+                else:
+                    sdlog.info("Required reboot pending, launching updater")
+                    return True
+            else:
+                sdlog.info(
+                    "Update status is {}, launching updater.".format(
```

Comment:
This results in log lines like "Update status is 1, launching updater" which is of course true, but is only useful to anyone familiar with the implementation internals -- I think for states we know the updater can enter, we should have corresponding human-readable log lines.

Reply:
:heavy_check_mark: